src/test/java/formflow/library/repository/SubmissionRepositoryServiceTest.java

```diff
@@ -38,6 +45,60 @@ void shouldSaveASubmissionWithUUID() {
     assertThat(firstSubmission.getId()).isInstanceOf(UUID.class);
   }
 
+  @Test
+  void testShortCodePersistsOneTimeOnly() {
+    Submission submission = new Submission();
+    submission.setFlow("testFlow");
+    assertThat(submission.getShortCode()).isNull();
+
+    submission = saveAndReload(submission);
+    assertThat(submission.getId()).isInstanceOf(UUID.class);
+    assertThat(submission.getShortCode()).isNotNull();
+
+    // application-test.yaml sets this to 8, to override the default behavior
+    // this just tests that the config is indeed working and the default of 6 is not used
+    assertThat(submission.getShortCode().length()).isEqualTo(8);
+    assertThat(submission.getShortCode().matches("[A-Za-z0-9]+")).isEqualTo(true);
+
+    Optional<Submission> reloadedSubmission = submissionRepositoryService.findById(submission.getId());
+    if (reloadedSubmission.isPresent()) {
+      assertThat(submission.getShortCode()).isEqualTo(reloadedSubmission.get().getShortCode());
+
+      Optional<Submission> reloadedSubmission2 = submissionRepositoryService.findById(submission.getId());
+      if (reloadedSubmission2.isPresent()) {
+        assertThat(reloadedSubmission2.get().getShortCode()).isEqualTo(reloadedSubmission.get().getShortCode());
+      } else {
+        Assertions.fail();
+      }
+    } else {
+      Assertions.fail();
+    }
+  }
+
+  @Test
+  void testFindByShortCode() {
+    Submission submission = new Submission();
+    submission.setFlow("testFlow");
+    assertThat(submission.getShortCode()).isNull();
+
+    submission = saveAndReload(submission);
+    assertThat(submission.getId()).isInstanceOf(UUID.class);
+    assertThat(submission.getShortCode()).isNotNull();
+
+    // application-test.yaml sets this to 8, to override the default behavior
+    // this just tests that the config is indeed working and the default of 6 is not used
+    assertThat(submission.getShortCode().length()).isEqualTo(8);
+    assertThat(submission.getShortCode().matches("[A-Za-z0-9]+")).isEqualTo(true);
+
+    Optional<Submission> reloadedSubmission = submissionRepositoryService.findByShortCode(submission.getShortCode());
+    if (reloadedSubmission.isPresent()) {
+      assertThat(submission).isEqualTo(reloadedSubmission.get());
+      assertThat(submission.getShortCode()).isEqualTo(reloadedSubmission.get().getShortCode());
+    } else {
+      Assertions.fail();
+    }
+  }
+
```

Comment:
Maybe some tests for each of the different configurations? Annoyingly I haven't figured out a good way to load different properties per test without the need for multiple test files :(

Reply:
Added a few tests in new files that handle the permutations of configs. I think I should also figure out how to add this to any tests that actually hit the ScreenController too.